### PR TITLE
Vulnerabilities update

### PR DESCRIPTION
**Pull request by Cheshiriks**
- added _shortDescription_, _relatedLink_, _language_ fields in Vulnerability entity and VulnerabilityDto
- added _type_ field and remove _isOpenSource_ field in VulnerabilityProject entity and VulnerabilityProjectDto
- added _VulnerabilityLanguage_ and _VulnerabilityProjectType_ enum classes
- changed _createVulnerabilityView_
- changed _fossGraphCollectionView_
- changed _vulnerabilityProjectWindow_

### CODE REVIEW

File: save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/entities/vulnerability/VulnerabilityLanguage.kt [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+@Serializable
+@JsExport
+@Suppress("IDENTIFIER_LENGTH")
+enum class VulnerabilityLanguage(val value: String) {
+    C("C"),
+    CPP("Cpp"),
+    JAVA("Java"),
+    JS("JS"),
+    KOTLIN("Kotlin"),
+    NONE("None"),
```
Comment (sanyavertolet) [inline save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/entities/vulnerability/VulnerabilityLanguage.kt:19]: Isn't it better to add (or even replace `NONE` with) `OTHER`?

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/fossgraph/CreateVulnerabilityView.kt [hunk 2]
```diff
@@ -142,6 +173,47 @@ val createVulnerabilityView: VFC = VFC {
...
+                                    }
+                                }
+
+                                div {
+                                    className = ClassName("col-md-12 pl-2 pr-2 mt-3 text-left align-self-center")
+                                    +"Language"
+
+                                    div {
+                                        className = ClassName("input-group")
+                                        ReactHTML.select {
```
Comment (sanyavertolet) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/fossgraph/CreateVulnerabilityView.kt:193]: again and again :)

Why do you use `select` btw? We have `selectBuilder` or smth like that

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/fossgraph/CreateVulnerabilityView.kt [hunk 3]
```diff
@@ -142,6 +173,47 @@ val createVulnerabilityView: VFC = VFC {
...
+                                            className = ClassName("form-control")
+                                            VulnerabilityLanguage.values().map { it.toString() }.forEach {
+                                                ReactHTML.option {
```
Comment (sanyavertolet) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/fossgraph/CreateVulnerabilityView.kt:196]:
```suggestion
                                                option {
```

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/fossgraph/VulnerabilityProjectWindow.kt [hunk 4]
```diff
@@ -90,15 +91,25 @@ private fun createVulnerabilityProjectWindow() = FC<VulnerabilityProjectWindowPr
...
-                id = "checkbox"
-                defaultChecked = false
-                onChange = { event ->
-                    setVulnerabilityProject(vulnerabilityProject.copy(isOpenSource = event.target.checked))
+            className = ClassName("col-md-12 pl-2 pr-2 mt-3 text-left align-self-center")
+            +"Type of project: "
+
+            div {
+                className = ClassName("input-group")
+                select {
```
Comment (sanyavertolet) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/fossgraph/VulnerabilityProjectWindow.kt:99]: Why not `selectBuilder`?

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/fossgraph/CreateVulnerabilityView.kt (hunk 3 above)
Comment (Cheshiriks) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/fossgraph/CreateVulnerabilityView.kt]: Fixed

**Review by nulls — APPROVED** [review]

**Pull request merged**